### PR TITLE
Unistaller 2/2

### PR DESCRIPTION
**Pull request by palexster**
# Description

This PR completes the uninstall process by waiting for the effective unjoin (incoming and outgoing)

### CODE REVIEW

File: pkg/uninstaller/wait.go [hunk 1]
```diff
@@ -0,0 +1,156 @@
...
+	quit := make(chan struct{})
+	var wError error
+	value := false
+	var res = &resultType{
+		Resource: toCheck,
+		Success:  false,
+	}
+	for {
+		select {
+		case <-timeout.C:
```
Comment (aleoli) [inline pkg/uninstaller/wait.go:135]: Is it possible to factorize this behavior with the very similar one a few lines before?

**Review by aleoli — APPROVED** [review]

File: pkg/uninstaller/wait.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	"k8s.io/klog/v2"
+	"sync"
+	"time"
+)
+
+// WaitForResources waits until existing peerings are disabled and associated resources are removed.
+func WaitForResources(client dynamic.Interface) error {
+	klog.Info("Waiting for Liqo Resources to be correctly deleted")
+	var wg sync.WaitGroup
+	DeletionResult := make(chan *resultType, len(toCheck))
```
Comment (mlavacca) [inline pkg/uninstaller/wait.go:18]: Why did you export this variable?

File: pkg/uninstaller/wait.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	ConditionResult := make(chan *resultType, ConditionsToCheck)
```
Comment (mlavacca) [inline pkg/uninstaller/wait.go:19]: DITTO!

File: pkg/uninstaller/helpers.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+	}
+	return foreign, nil
+}
+
+// checkPeeringsStatus verifies if all clusters have not active peerings
+func checkPeeringsStatus(foreign *discoveryV1alpha1.ForeignClusterList) bool {
+	var returnValue = true
+	for _, item := range foreign.Items {
+		if item.Status.Incoming.Joined || item.Status.Outgoing.Joined {
+			klog.Infof("Cluster %s have still a valid peering: (Incoming: %s, Outgoing: %s", item.Name, item.Status.Incoming.Joined, item.Status.Outgoing.Joined)
```
Comment (giorio94) [inline pkg/uninstaller/helpers.go:33]:
```suggestion
			klog.Infof("Cluster %s still has a valid peering: (Incoming: %s, Outgoing: %s", item.Name, item.Status.Incoming.Joined, item.Status.Outgoing.Joined)
```

**Review by giorio94 — APPROVED** [review]
/lgtm

**Pull request merged**